examples/01_random_field/07_precipitation.py

```diff
@@ -0,0 +1,114 @@
+"""
+Creating Synthetic Precipitation Fields
+---------------------------------------
+
+In this example we want to create a time series of a synthetic precipitation
+field.
+
+We'll start off by creating a Gaussian random field with an exponential
+variogram, which seems to reproduce the spatial correlations of precipitation
+fields quite well. We'll create a daily timeseries over a one dimensional cross
+section of 50km. This workflow is suited for sub daily precipitation time
+series.
+"""
+
+import copy
+import numpy as np
+import matplotlib.pyplot as plt
+import gstools as gs
+
+# fix the seed for reproducibility
+seed = 20170521
+# half daily timesteps over three months
+t = np.arange(0., 90., 0.5)
+# spatial axis of 50km with a resolution of 1km
+x = np.arange(0, 50, 1.0)
+
+# an exponential variogram with a corr. lengths of 2d and 5km
+model = gs.Exponential(dim=2, var=1.0, len_scale=2., anis=2.5)
+# create a spatial random field instance
+srf = gs.SRF(model)
+
+# a Gaussian random field which is also saved internally for the transformations
+srf.structured((t, x), seed=seed)
+P_gau = copy.deepcopy(srf.field)
+
+###############################################################################
+# Now we should take care of the dry periods. Therefore we simply introduce a
+# lower threshold value.
+
+threshold = 0.4
+srf.field[srf.field <= threshold] = 0.0
+P_cut = srf.field
+
+###############################################################################
+# With the above lines of code we have created a cut off Gaussian spatial
+# random field with an exponential variogram. But precipitation fields are not
+# distributed Gaussian. Thus, we will now transform the field with a box-cox
+# transformation, which is often used to account for the skewness of
+# precipitation fields. Different values have been suggested for the
+# transformation parameter lambda, but we will stick to 1/2. We call the
+# resulting field Gaussian anamorphosis.
+
+gs.transform.boxcox(srf, lmbda=0.5, shift=-1.0)
+
+###############################################################################
+# As a last step, the amount of precipitation is set. This should of course be
+# calibrated towards observations (the same goes for the threshold).
+
+amount = 3.0
+srf.field *= amount
+P_ana = srf.field
+
+###############################################################################
+# Finally we can have a look at the fields resulting from each step. For a
+# closer look, we will examine a cross section at an arbitrary location. And
+# afterwards we will create a contour plot for visual candy.
+
+fig, axs = plt.subplots(2, 2, sharex=True, sharey=True)
+
+axs[0,0].set_title('Gaussian')
+axs[0,0].plot(t, P_gau[:, 20])
+axs[0,0].set_ylabel(r'$P$ / mm')
```

Comment:
I was a bit confused by the `/` as separation for the unit. What do you think about `P [mm]`?

Reply:
It's the best and mathematically most rigorous way of labeling axes! Let me explain:
E.g., you read from the axes and the graph at a certain point in time: _P_ / mm = 40 You divide the equation by the unit mm and get
_P_ = 40mm
tada!

But yeah, I had to already convince other people, like coauthors of this way of life, as I like to call it ;-)

Reply:
Sneaky! Ok, you got me. Like it! Will do so too from now on! :+1: